src/views/datasources/dataimportcron/hooks/useDataImportCronActions.tsx

```diff
@@ -4,28 +4,77 @@ import { useHistory } from 'react-router-dom';
 import DataImportCronModel, {
   DataImportCronModelRef,
 } from '@kubevirt-ui/kubevirt-api/console/models/DataImportCronModel';
-import { V1beta1DataImportCron } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import DataSourceModel from '@kubevirt-ui/kubevirt-api/console/models/DataSourceModel';
+import {
+  V1beta1DataImportCron,
+  V1beta1DataSource,
+} from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
 import { AnnotationsModal } from '@kubevirt-utils/components/AnnotationsModal/AnnotationsModal';
 import DeleteModal from '@kubevirt-utils/components/DeleteModal/DeleteModal';
 import { LabelsModal } from '@kubevirt-utils/components/LabelsModal/LabelsModal';
 import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
 import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
 import { asAccessReview } from '@kubevirt-utils/resources/shared';
-import { Action, k8sDelete, k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
-import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk-internal';
+import { Action, k8sDelete, k8sGet, k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import { Loading } from '@patternfly/quickstarts';
+import { Split, SplitItem } from '@patternfly/react-core';
 
-type UseDataImportCronActionsProvider = (DataImportCron: V1beta1DataImportCron) => Action[];
+import { isDataResourceOwnedBySSP } from '../../utils';
+import { DataImportCronManageModal } from '../details/DataImportCronManageModal/DataImportCronManageModal';
+
+type UseDataImportCronActionsProvider = (
+  DataImportCron: V1beta1DataImportCron,
+) => [actions: Action[], onOpen: () => void];
 
 export const useDataImportCronActionsProvider: UseDataImportCronActionsProvider = (
   dataImportCron,
 ) => {
-  const [namespace] = useActiveNamespace();
+  const dataSourceName = dataImportCron?.spec?.managedDataSource;
+  const [dataSource, setDataSource] = React.useState<V1beta1DataSource>();
+  const [isLoading, setIsLoading] = React.useState(true);
   const { t } = useKubevirtTranslation();
   const { createModal } = useModal();
   const history = useHistory();
+  const isOwnedBySSP = isDataResourceOwnedBySSP(dataImportCron);
+
+  const lazyLoadDataSource = React.useCallback(async () => {
+    if (!dataSource && !isOwnedBySSP) {
+      setIsLoading(true);
+      const lazyDataSource = await k8sGet<V1beta1DataSource>({
+        model: DataSourceModel,
+        name: dataSourceName,
+        ns: dataImportCron?.metadata?.namespace,
+      });
+      lazyDataSource?.kind !== 'DataSourceList' && setDataSource(lazyDataSource);
```

Comment:
Why this? In which conditions we get a `DataSourceList` ?

Reply:
weird behavior, if k8sGet doesn't find the specific resource, it returns the whole list 😕